runtime/sema/check_variable_declaration.go

```diff
@@ -173,6 +199,11 @@ func (checker *Checker) visitVariableDeclarationValues(declaration *ast.Variable
 				declaration.SecondTransfer,
 				true,
 			)
+
+			if recordedResourceInvalidation != nil {
+				checker.resources.RemoveTemporaryMoveInvalidation(recordedResourceInvalidation.resource, recordedResourceInvalidation.invalidation)
+			}
+			checker.VisitExpression(declaration.Value, expectedValueType)
```

Comment:
Could we do something similar to the swap statement is checked, where the expression is visited once as the target/lhs of as assignment (using `checker.visitAssignmentValueType(declaration.Value)`), and then the next time it is visited as the value-expression/rhs?

Reply:
Yes, technically, but because the expression is necessarily a variable, member or index expression, these two checks are semantically identical.